lib/trip.rb

```diff
@@ -1,66 +1,86 @@
-require 'csv'
-
-require_relative 'csv_record'
+require "csv"
+require "time"
+require_relative "csv_record"
 
 module RideShare
   class Trip < CsvRecord
-    attr_reader :id, :passenger, :passenger_id, :start_time, :end_time, :cost, :rating
+    attr_reader :passenger, :passenger_id, :start_time, :end_time, :cost, :rating, :driver_id, :driver
 
     def initialize(
-          id:,
-          passenger: nil,
-          passenger_id: nil,
-          start_time:,
-          end_time:,
-          cost: nil,
-          rating:
-        )
+      id:,
+      passenger: nil,
+      passenger_id: nil,
+      start_time:,
+      end_time:,
+      cost: nil,
+      rating: nil,
+      driver_id: nil,
+      driver: nil
+    )
       super(id)
 
       if passenger
         @passenger = passenger
         @passenger_id = passenger.id
-
       elsif passenger_id
         @passenger_id = passenger_id
+      else
+        raise ArgumentError, "Passenger or passenger_id is required"
+      end
 
+      if driver
+        @driver = driver
+        @driver_id = driver.id
+      elsif driver_id
+        @driver_id = driver_id
       else
-        raise ArgumentError, 'Passenger or passenger_id is required'
+        raise ArgumentError, "Driver or driver_id is required"
       end
 
       @start_time = start_time
       @end_time = end_time
       @cost = cost
       @rating = rating
 
-      if @rating > 5 || @rating < 1
-        raise ArgumentError.new("Invalid rating #{@rating}")
+      if @rating != nil
+        if @rating > 5 || @rating < 1
+          raise ArgumentError.new("Invalid rating #{@rating}")
+        end
+      end
+
+      if @end_time != nil
+        raise ArgumentError.new("The start time should be before the end time") if @end_time < @start_time
       end
```

Comment:
This can be simplified like above:

```suggestion
      raise ArgumentError.new("The start time should be before the end time") if @end_time && @end_time < @start_time
```